modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java

```diff
@@ -85,13 +88,22 @@ protected void doParse(Builder definitionBuilder) throws ConfigurationException
     type.getFields().forEach(this::parseField);
   }
 
+  private String getFieldName(ObjectFieldType objectFieldType, String fieldName) {
+
+    return objectFieldType.getValue().getAnnotation(TypeIdAnnotation.class)
```

Comment:
You will end up needing the .orElse() to deal with the case in which the objectField doesn't have the TypeIdAnnotation.